src/audio/google/google_rtc_audio_processing.c

```diff
@@ -563,125 +591,191 @@ static int google_rtc_audio_processing_reset(struct processing_module *mod)
 	return 0;
 }
 
+static int16_t convert_float_to_uint16_hifi(float data)
+{
+	const xtfloat ratio = 2 << 14;
+	xtfloat x0 = data;
+	xtfloat x1;
+	int16_t x;
+
+	x1 = XT_MUL_S(x0, ratio);
+	x = XT_TRUNC_S(x1, 0);
+
+	return x;
+}
+
+static float convert_uint16_to_float_hifi(int16_t data)
+{
+	const xtfloat ratio = 2 << 14;
+	xtfloat x0 = data;
+	float x;
+
+	x = XT_DIV_S(x0, ratio);
+
+	return x;
+}
```

Comment:
AFAIK we usually make HiFi code depend on availability devinitions perhaps something like `XCHAL_HAVE_HIFI3` and also provide pure C alternatives

Reply:
no need here, this is a shmm for 3rd party lib that is using HiFI and will always be compiled using clang

Reply:
> using HiFI and will always be compiled using clang

HiFI and clang are two totally different things, don't mix them up, it's confusing. For instance this #8637 _requires_ clang (it literally can't compile with GCC) and it absolutely can't use HiFI.

I  think you mean "closed-source Cadence Xtensa toolchain" (or any shorter version of this.

Reply:
> I think you mean "compiled only by closed-source Cadence Xtensa toolchain" (or any shorter version of this).

And this is in fact wrong:

https://github.com/thesofproject/sof/actions/runs/7226339953/job/19691685718?pr=8637

```
[208/259] Building modules_sof.dir/home/workspace/sof/src/audio/drc/drc_generic.c.obj
[209/259] Building modules_sof.dir/home/workspace/sof/src/audio/google/google_rtc_audio_processing_mock.c.obj
[210/259] Building modules_sof.dir/home/workspace/sof/src/audio/igo_nr/igo_nr_stub.c.obj
[211/259] Building modules_sof.dir/home/workspace/sof/src/audio/multiband_drc/multiband_drc.c.obj
[212/259] Building modules_sof.dir/home/workspace/sof/src/audio/google/google_rtc_audio_processing.c.obj

```